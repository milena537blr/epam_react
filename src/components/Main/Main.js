import React, { Component } from "react";
import PropTypes from "prop-types";
import s from "./Main.module.scss";
import Box from "../Box/Box";
import Card from "../Card/Card";

class Main extends Component {
  constructor(props) {
    super(props);
    this.state = {
      filterText: ""
    };
  }

  render() {
    const cards = this.props.cards.map(card => {
      return (
        <Card
          cardClick={this.props.onHandleCardClick}
          card={card}
          key={card.id}
        />
      );
    });

    return (
      <main className={s.main}>
        <div className={s.container}>
          <Box className={s.wrapper} align="center">
            {cards}
          </Box>
        </div>
      </main>
    );
  }
}

Main.propTypes = {
  cards: PropTypes.array,
  searchText: PropTypes.string,
  onHandleCardClick: PropTypes.func
};

export default Main;
