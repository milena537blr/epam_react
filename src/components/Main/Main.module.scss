@import "../../styles/core.scss";

.main {
  background-color: $color--white;
}

.container {
  @include container;
}

.wrapper {
  padding: rem(40) rem(20);
  display: flex;
  flex-wrap: wrap;
}
