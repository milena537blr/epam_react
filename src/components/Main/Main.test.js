import React from "react";
import { shallow } from "enzyme";
import Main from "./Main";
import renderer from "react-test-renderer";

// import data from "../../data/data";

const cards = [
  {
    id: 1,
    date: 2007,
    description: "American drama",
    genre: "Dramas",
    name: "Vampire diaries",
    text:
      "The Vampire Diaries is an American supernatural teen drama television series developed by Kevin Williamson and Julie Plec, based on the popular book series of the same name written by L. J. Smith.",
    time: 145
  },
  {
    id: 2,
    date: 2014,
    description: "Drama/Comedy-drama",
    genre: "Drama",
    name: "The Fault in Our Stars",
    text:
      "Hazel Grace Lancaster (Shailene Woodley), a 16-year-old cancer patient, meets and falls in love with Gus Waters (Ansel Elgort), a similarly afflicted teen from her cancer support group. Hazel feels that Gus really understands her. ",
    time: 133
  }
];

describe("<Main>", () => {
  it("renders a `.main`", () => {
    const wrapper = shallow(<Main cards={cards} />);
    expect(wrapper.find(".main")).toExist();
  });

  it('<Main> snapshot test', () => {
    const component = renderer.create(<Main cards={cards} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
