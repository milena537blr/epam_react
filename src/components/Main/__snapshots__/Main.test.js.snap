// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main> <Main> snapshot test 1`] = `
<main
  className="main"
>
  <div
    className="container"
  >
    <div
      className="root wrapper horizontal center"
      style={
        Object {
          "height": "undefined",
          "margin": "undefined",
          "marginBottom": "undefined",
          "marginLeft": "undefined",
          "marginRight": "undefined",
          "marginTop": "undefined",
          "maxHeight": "undefined",
          "maxWidth": "undefined",
          "minHeight": "undefined",
          "minWidth": "undefined",
          "padding": "undefined",
          "paddingBottom": "undefined",
          "paddingLeft": "undefined",
          "paddingRight": "undefined",
          "paddingTop": "undefined",
          "width": "undefined",
        }
      }
    >
      <figure
        className="card"
      >
        <button
          area-label="Vampire diaries"
          onClick={[Function]}
        >
          <img
            alt="Vampire diaries"
            className="image"
            src="test-file-stub"
          />
        </button>
        <figcaption>
          <div
            className="root horizontal spaceBetween middle"
            style={
              Object {
                "height": "undefined",
                "margin": "undefined",
                "marginBottom": "undefined",
                "marginLeft": "undefined",
                "marginRight": "undefined",
                "marginTop": "undefined",
                "maxHeight": "undefined",
                "maxWidth": "undefined",
                "minHeight": "undefined",
                "minWidth": "undefined",
                "padding": "undefined",
                "paddingBottom": "undefined",
                "paddingLeft": "undefined",
                "paddingRight": "undefined",
                "paddingTop": "undefined",
                "width": "undefined",
              }
            }
          >
            <div
              className="name"
            >
              Vampire diaries
            </div>
            <div
              className="date"
            >
              2007
            </div>
          </div>
          <div
            className="genre"
          >
            Dramas
          </div>
        </figcaption>
      </figure>
      <figure
        className="card"
      >
        <button
          area-label="The Fault in Our Stars"
          onClick={[Function]}
        >
          <img
            alt="The Fault in Our Stars"
            className="image"
            src="test-file-stub"
          />
        </button>
        <figcaption>
          <div
            className="root horizontal spaceBetween middle"
            style={
              Object {
                "height": "undefined",
                "margin": "undefined",
                "marginBottom": "undefined",
                "marginLeft": "undefined",
                "marginRight": "undefined",
                "marginTop": "undefined",
                "maxHeight": "undefined",
                "maxWidth": "undefined",
                "minHeight": "undefined",
                "minWidth": "undefined",
                "padding": "undefined",
                "paddingBottom": "undefined",
                "paddingLeft": "undefined",
                "paddingRight": "undefined",
                "paddingTop": "undefined",
                "width": "undefined",
              }
            }
          >
            <div
              className="name"
            >
              The Fault in Our Stars
            </div>
            <div
              className="date"
            >
              2014
            </div>
          </div>
          <div
            className="genre"
          >
            Drama
          </div>
        </figcaption>
      </figure>
    </div>
  </div>
</main>
`;
